src/kifshare/ranges.clj

```diff
@@ -0,0 +1,89 @@
+(ns kifshare.ranges
+  (:require [clojure.tools.logging :as log]
+            [kifshare.inputs :refer [chunk-stream]]))
+
+(defn range-request?
+  [ring-request]
+  (and (contains? ring-request :headers)
+       (contains? (:headers ring-request) "range")))
+
+;; begins with bytes=; then a start-end range (where excluding either number works, but not both)
+;; regex lookahead/lookbehind to ensure only one number is missing, not both
+(def ^:private range-regex #"\s*(bytes)\s*=\s*([0-9]+|(?=-\d))\-([0-9]+|(?<=\d-))\s*")
```

Comment:
A lot is buried in this regex but it works better than the anon-files version did. I updated this when updating the range support in kifshare more generally a few weeks ago.